platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,346 @@
+proposal only introduces the logistics of generating the key material and the logic for deciding to adopt the CAB in
+replacement of the AAB.
+
+#### TSS Algorithm
+
+The threshold signature scheme chosen for our implementation is detailed in https://eprint.iacr.org/2021/339. This
+scheme is able to meet the core requirements listed above. The rest of this proposal details the design of how to
+implement this scheme in the consensus nodes.
+
+The Groth21 algorithm uses Shamir Secret Sharing to hide the private key of the ledger and distribute shares of it
+so that a threshold number of node signatures on a message can generate the ledger signature on the same message.  
+The threshold should be high enough to ensure at least 1 honest node is required to participate in the signing since
+a threshold number of nodes can also collude to recover the ledger private key, if they so choose. For example if
+the network is designed to tolerate up to 1/3 of the nodes being dishonest, then the threshold should never be less
+than 1/3 + 1 of the nodes to ensure that no 1/3 will collude to recover the ledger private key.
+
+The Groth21 algorithm requires specific Elliptic Curve (EC)s with the ability to produce bilinear pairings. Each
+node will need its own long-term EC key pair on the curve for use in the groth21 TSS algorithm. Each node receives
+some number of shares in proportion to the consensus weight assigned to it. Each share is an EC key pair on the
+same curve. All public keys of the shares are known, and aggregating a threshold number of share public keys will
+produce the ledger id. Only the node has access to the private key of its own shares. Each node must keep their
+private keys to themselves since a threshold number of share private keys can aggregate to recover the ledger
+private key. An aggregate number of signatures on the same message from the share private keys will aggregate to
+recover the ledger signature on the message.
+
+Transferring the ability to generate ledger signatures from one set of consensus nodes to another is done by having
+each node generate a `TssMessage` for each share that fractures the share into a number of subshares or `shares of
+shares`. Each `Share of a share` is encrypted with the public key of the node in the next consensus roster that it
+belongs to so that only the intended node can use the `share of a share` to recover that node's share's private key.  
+The collection of `TssMessages` generated from nodes in the previous consensus roster forms the `key material` for
+the new consensus roster. To bootstrap a network and generate the ledger private and public keys, each node creates
+a random share for themselves and generates a `TssMessage` containing `shares of shares` for the total number of shares
+needed in the consensus roster. The use of random shares at the outset creates a random ledger private key that nobody
```

Comment:
Again a diagram would help a lot. Your text is really well written, but it is a super complexe topic - at least for me ;)

Reply:
adding diagram.